dockerfile_scripts/install_libnss_determined.sh

```diff
@@ -5,5 +5,19 @@ set -e
 # Add a plugin to the user system that lets us extend the users available in
 # the container at runtime. This is critical for supporting non-root shell,
 # which in turn is critical for non-root distributed training.
+#
+# Also, to work around a behavior in OpenShift 4.+ [1], place `determined`
+# NSS plug-in to be searched first instead of last so Determined-defined
+# entries are used when there are conflicts with infrastructure-based
+# entries in these identity databases: passwd, shadow, group.
+#
+# [1] https://cloud.redhat.com/blog/a-guide-to-openshift-and-uids "By default,
+# OpenShift 4.x appends the effective UID into /etc/passwd of the Container
+# during the creation of the Pod."
+
 make -C /tmp/det_dockerfile_scripts/libnss_determined libnss_determined.so.2 install \
-    && sed -E -i -e 's/^((passwd|shadow|group):.*)/\1 determined/' /etc/nsswitch.conf
+    && sed -i -E '
+        /^(passwd|group|shadow):/{
+            s/[ \t]determined//;
+            s/(:[ \t]+)/\1determined /;
```

Comment:
suggested simplification: `s/:/: determined/`

Reply:
```
passwd:         files systemd
group:          files systemd
shadow:         files
gshadow:        files
```

.. becomes

```
passwd: determined         files systemd
group: determined          files systemd
shadow: determined         files
gshadow:        files
```

I really don't mind.  I was merely preserving appearance, but who's gonna be looking at `nsswitch.conf` in an ephemeral container anyway?